cmd/frontend/backend/versions.go

```diff
@@ -0,0 +1,96 @@
+package backend
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/keegancsmith/sqlf"
+	"github.com/sourcegraph/sourcegraph/internal/db/dbconn"
+	"github.com/sourcegraph/sourcegraph/internal/db/dbutil"
+)
+
+// UpgradeError is returned by UpdateServiceVersion when it faces an
+// upgrade policy violation error.
+type UpgradeError struct {
+	Service  string
+	Previous *semver.Version
+	Latest   *semver.Version
+}
+
+// Error implements the error interface.
+func (e UpgradeError) Error() string {
+	return fmt.Sprintf(
+		"upgrading %q from %q to %q is not allowed, please refer to %s",
+		e.Service,
+		e.Previous,
+		e.Latest,
+		"https://docs.sourcegraph.com/#upgrading-sourcegraph",
+	)
+
+}
+
+// UpdateServiceVersion updates the latest version for the given Sourcegraph
+// service. It enforces our documented upgrade policy.
+// https://docs.sourcegraph.com/#upgrading-sourcegraph
+func UpdateServiceVersion(ctx context.Context, service, version string) error {
+	latest, err := semver.NewVersion(version)
```

Comment:
nit:
```suggestion
	current, err := semver.NewVersion(version)
```

It's not necessarily true that a customer is always performing upgrade to the latest version.